orbit/utils/kernels.py

```diff
@@ -82,3 +84,73 @@ def sandwich_kernel(x, x_i):
     k = k / np.sum(k, axis=1, keepdims=True)
 
     return k
+
+def generate_tp(prediction_date_array, training_df_meta):
```

Comment:
I wonder should we put the code here.  I would expect kernels are the places to provide different choices of kernels.  The code you put seems more like utilities and may be specifically for KTR.  If would suggest a separate file to store this.  Ultimately, we should re-visit the usefulness of them and put the useful one back as a static method sounds better.